### PR TITLE
[TODO] Blockers @okdas

### PR DESCRIPTION
**Pull request by okdas**
## Summary

Resolving a few blockers I've had mentioned in the codebase.

## Issue

- #587

## Type of change

Select one or more:

- [ ] New feature, functionality or library
- [ ] Bug fix
- [x] Code health or cleanup
- [ ] Documentation
- [ ] Other (specify)

## Testing

**Documentation changes** (only if making doc changes)
- [ ] `make docusaurus_start`; only needed if you make doc changes

**Local Testing** (only if making code changes)
- [x] **Unit Tests**: `make go_develop_and_test`
- [x] **LocalNet E2E Tests**: `make test_e2e`
- See [quickstart guide](https://dev.poktroll.com/developer_guide/quickstart) for instructions

**PR Testing** (only if making code changes)
- [ ] **DevNet E2E Tests**: Add the `devnet-test-e2e` label to the PR.
    - **THIS IS VERY EXPENSIVE**, so only do it after all the reviews are complete.
    - Optionally run `make trigger_ci` if you want to re-trigger tests without any code changes
    - If tests fail, try re-running failed tests only using the GitHub UI as shown [here](https://github.com/pokt-network/poktroll/assets/1892194/607984e9-0615-4569-9452-4c730190c1d2)

## Sanity Checklist

- [x] I have tested my changes using the available tooling
- [x] I have commented my code
- [x] I have performed a self-review of my own code; both comments & source code
- [ ] I create and reference any new tickets, if applicable
- [x] I have left TODOs throughout the codebase, if applicable

## Summary by CodeRabbit

- **Chores**
  - Enabled additional linters and updated linter settings for code quality checks.
  - Updated `golangci-lint` version in the `Makefile` to the latest version.
  - Improved error handling and logging across multiple files.
  - Simplified and streamlined code for better readability and maintainability.
  - Updated documentation comments and added clarifying comments in the code.
  - Removed deprecated functions and replaced them with updated alternatives.

### CODE REVIEW

**Comment by okdas** [comment]
Skipping e2e test as no go code changes were made. I did verify the LocalNet set up though, as the Tiltfile was changed.

File: x/shared/helpers/service_configs.go
Comment (bryanchriswhite) [inline x/shared/helpers/service_configs.go:73]: cc @red-0ne

**Review by bryanchriswhite — COMMENTED** [review]
Nice one @okdas - thanks for picking this back up! :fire: 

~~What was your thinking on the order of operations to getting CI passing on main?~~

EDIT: I opened a PR with the linter fixes. I figure once it asand this PR have approval, we can merge it into this branch prior to merging this with main. Wdyt?
- #684

**Review by bryanchriswhite — APPROVED** [review]

File: x/shared/helpers/service_configs.go
Comment (okdas) [reply x/shared/helpers/service_configs.go:73]: @bryanchriswhite had a chat with @red-0ne [in Discord](https://discord.com/channels/824324475256438814/1138895490331705354/1260969507456024596) so I'll adjust the comment accordingly.

**Comment by okdas** [comment]
@bryanchriswhite since you've reviewed this PR and also #684, I'm going to merge this.

One note I want to add is I had to also add extra changes in [this commit](https://github.com/pokt-network/poktroll/pull/674/commits/f3b972ee41f6b06a3827ec941d41fe5b377641e2) to make linter happy. Please let me know if any of these do not make sense.

**Pull request merged**
